fvm/errors/todo_rm_emulator_hack.go

```diff
@@ -23,10 +23,10 @@ type InvalidEnvelopeSignatureError struct{ CodedError }
 //type InvalidProposalSeqNumberError struct{ CodedError }
 
 // DO NOT USE.
-//type InvalidPayloadSignatureError struct{ CodedError }
+type InvalidPayloadSignatureError struct{ CodedError }
```

Comment:
what is this file for?

And it has a comment says "DO NOT USE"

Reply:
(See the comment at the top of the file)

We need to keep the original struct around because we're in circular dependency hell.  The flow-go integration test depends on the emulator, which in turn depends on flow-go.  We can't delete fvm structs / apis until we've upgraded the emulator, and we can't do that until after a spork.  fvm is littered with 10+ similar TODOs =...(

Reply:
btw, for additional context, the emulator uses these struct for error classification.  But the checks are done incorrectly and the switch cases always ends up in the default branch.  I'm not breaking anything that wasn't already broken